### PR TITLE
MVP2-626 - TRADING ENGINE: drop the isolation level

### PR DESCRIPTION
**Pull request by SamirHodzic**

### CODE REVIEW

File: erc20_moc/Erc20.noblock.js [hunk 1]
```diff
@@ -72,38 +86,45 @@ class ERC20 {
...
+      const receiverBalanceRes = await updateBalanceOfUser(
+        dbClient,
+        receiver,
+        this.symbol,
+        trx_time,
+        amount
+      );
+      const receiverBalance = this._balanceOf(receiver, receiverBalanceRes);
+
+      if (receiverBalance < 0n) {
```
Comment (rudolfix) [inline erc20_moc/Erc20.noblock.js]: we can drop it. balance of receiver can never be negative. i know we discussed to keep it but later we decided we do not do any tricks with negative transfer amounts. and if they are not allowed then we are fine

File: erc20_moc/Erc20.noblock.js [hunk 2]
```diff
@@ -193,28 +214,29 @@ class ERC20 {
...
+      if (userBalance < 0n) {
         throw new NoWeb3Exception(
           `Owner can't burn more than it owns! -- Owner: ${sponsor} owns: ${balance} burns: ${amount}`
         );
       }
+
+      await insertTransaction(
+        dbClient,
+        sponsor,
+        '',
```
Comment (rudolfix) [inline erc20_moc/Erc20.noblock.js:228]: next version of trading engine will have NULL address properly defined. we can also allows nulls in the transaction table. I'll think about it

File: utils/db_helper.js [hunk 3]
```diff
@@ -93,18 +93,17 @@ const SET_CASINO_TRADE_OUTCOMES =
 const GET_CASINO_TRADES =
   'SELECT userId, crashFactor, stakedAmount FROM casino_trades WHERE gameId = $1 AND state = $2;';
 const SET_CASINO_TRADE_STATE =
-  'UPDATE casino_trades SET state = $1, crashfactor = $2 WHERE gameId = $3 AND state = $4 and userId = $5 RETURNING *;';
+  'UPDATE casino_trades SET state = $1, crashfactor = $2 WHERE gameId = (SELECT gameId FROM casino_trades WHERE gameId = $3 AND state = $4 AND userId = $5 FOR UPDATE) AND state = $4 AND userId = $5 RETURNING *;';
```
Comment (rudolfix) [inline utils/db_helper.js]: revert to old query

File: utils/db_helper.js [hunk 4]
```diff
@@ -76,7 +76,7 @@ const GET_BET_INVESTORS =
   'SELECT buyer, direction, SUM(investmentamount) AS amount FROM amm_interactions WHERE bet = $1 GROUP BY buyer, direction;';
 
 const UPDATE_BALANCE_OF_USER =
-  'INSERT INTO token_balances (owner, symbol, last_update, balance) VALUES($1, $2, $3, $4) ON CONFLICT (owner, symbol) DO UPDATE SET last_update = $3, balance = $4;';
+  'INSERT INTO token_balances (owner, symbol, last_update, balance) VALUES($1, $2, $3, $4) ON CONFLICT (owner, symbol) DO UPDATE SET last_update = $3, balance = token_balances.balance + $4 RETURNING balance;';
```
Comment (rudolfix) [inline utils/db_helper.js:79]: check if RETURNING works for insert. or we have `undefined`

**Review by rudolfix — DISMISSED** [review]

File: utils/db_helper.js (hunk 4 above)
Comment (SamirHodzic) [reply utils/db_helper.js:79]: Both insert/update will always return balance, no way to get `undefined` value.

**Pull request merged**